bootstraptest/test_yjit.rb

```diff
@@ -2431,6 +2431,27 @@ def foo(required:, specified: 999, simple_default: 3, complex_default: "4".to_i)
   5.times.map { foo(specified: 2, required: 1) }.uniq
 }
 
+# cfunc kwargs
+assert_equal '{:foo=>123}', %q{
+  def foo(bar)
+    bar.store(:value, foo: 123)
+    bar[:value]
+  end
+
+  foo({})
+  foo({})
+}
+
+# variadic cfunc kwargs
+assert_equal '{:foo=>123}', %q{
+  def foo(bar)
+    bar.merge(foo: 123)
+  end
+
+  foo({})
+  foo({})
+}
```

Comment:
Could also use a test case with zero arguments as that's often a corner case.

Reply:
There isn't really a zero arguments case when passing kwargs (there's always at least one for the kwargs) but I added another non-variadic only-kwarg case and some other tests